asserts/account_key.go

```diff
@@ -115,6 +138,21 @@ func assembleAccountKey(assert assertionBase) (Assertion, error) {
 		return nil, err
 	}
 
+	// XXX: We should require name to be non-empty after backfilling existing assertions.
+	value, ok := assert.headers["name"]
+	if ok {
+		name, ok := value.(string)
+		if !ok {
+			return nil, fmt.Errorf(`"name" header must be a string`)
+		}
+		if len(name) == 0 {
+			return nil, fmt.Errorf(`"name" header should not be empty`)
+		}
+		if !ValidAccountKeyName.MatchString(name) {
+			return nil, fmt.Errorf(`"name" header contains invalid characters: %q`, name)
+		}
```

Comment:
Maybe extract this check to header_checks.go so it can be reused for account-key-request and other assertions in the future? It would be nice to explain what was expected better, but all I can think of doing is embedding the regexp pattern in the error ... which is a bit ugly.